### ISSUE TITLE
Implementing TextMate support for a DSL created using Xtext

### ISSUE THREAD

**Issue by SharkJ** [eclipse/xtext#1327]
Hi,
I have a simple DSL made using Xtext in Eclipse with the following grammar : -

```
grammar org.xtext.example.mydsl.MyDsl with org.eclipse.xtext.common.Terminals

generate myDsl "http://www.xtext.org/example/mydsl/MyDsl"

Model:
    greetings+=Greeting*;

Greeting:
    'Hello' name=ID '!';
```
When I run this as a plugin in Eclipse, it works fine with syntax highlighting. However, when I tried to run it as a Language Server in Eclipse Che, the syntax highlighting service did not work.
Is it possible to overcome this problem by implementing TextMate or any other approach? How should it be done?

Thanks!

**Comment by cdietrich** [comment]
Hi,

i am not sure what problem you try to adress:
- either che support lsp highlighting => work OOTB
- if not then che has another mechanism on how to do client side highlighting (as vscode, atom etc have = textmate grammars)
=> this is a che, not an xtext question

**Comment by SharkJ** [comment]
Hi,
Is there a way to implement a TextMate file with the Xtext grammar that I've created in Eclipse IDE?

**Comment by cdietrich** [comment]
No you have to do that manually.
This snippet might come you handy

public class Main {
	
	public static void main(String[] args) {
		Injector injector = new MyDslStandaloneSetup().createInjectorAndDoEMFRegistration();
		MyDslGrammarAccess ga = injector.getInstance(MyDslGrammarAccess.class);
		System.out.println(GrammarUtil.getAllKeywords(ga.getGrammar()));
	}
​
}

You might even write a own generator fragment and include it in the workflow
(And contribute it to Xtext)

**Comment by SharkJ** [comment]
Thanks! 
In [this repo](https://github.com/svenefftinge/xtext-lsp-workshop) the Xtext DSL is implemented with LSP4E, and a TextMate file is linked to the LSP4E project. Is there a possibility to combine the Xtext project and LSP4E project to get a useful output (a .jar file with all dependencies)? Or, is there a way to implement classes using the LSP4J files created when the Xtext DSL is built, in order to achieve my goal?

**Comment by cdietrich** [comment]
I don’t understand your question 
How to bundle client specific things is up to the client (in this case lsp4e) and in lsp4e everything’s packaged into the plugin (See Plugin.xml and build.properties)
I don’t see how that relates to Che

**Comment by SharkJ** [comment]
@cdietrich Sorry, I learnt that Eclipse Che gets support from Orion editor, so my attempt to add TextMate grammar is of no use, from what I understand.

> No you have to do that manually.
> This snippet might come you handy
> 
> public class Main {
> 
> ```
> public static void main(String[] args) {
> 	Injector injector = new MyDslStandaloneSetup().createInjectorAndDoEMFRegistration();
> 	MyDslGrammarAccess ga = injector.getInstance(MyDslGrammarAccess.class);
> 	System.out.println(GrammarUtil.getAllKeywords(ga.getGrammar()));
> }
> ```
> ​
> }

Could you please tell me to where in my project, could I make use of this snippet that you've provided. It would be of a great help.
Thanks!

**Comment by cdietrich** [comment]
simply drop it to the .mydsl or .mydsl.ide or mydsl.tests project

**Comment by SharkJ** [comment]
Thanks a lot for the support! It works. 
For anybody who would encounter the same problem, I created a Main.java file under org.xtext.hello/src/main/java/org.xtext.hello and included the code snippet provided by @cdietrich . I had to edit the code a bit to suit my language (since its name was not MyDsl, I had to simply replace every instance mentioning 'MyDsl' with my DSL's name ('Hello'). 
```
ex: -
Injector injector = new MyDslStandaloneSetup().createInjectorAndDoEMFRegistration();

was replaced by,
Injector injector = new HelloStandaloneSetup().createInjectorAndDoEMFRegistration();
```

Then right-clicked the org.xtext.hello project -> Run As -> Java Application, and the console will output all the keywords you have used in your DSL.
Thank You @cdietrich !
